### PR TITLE
939 standardise helm charts

### PR DESCRIPTION
**Pull request by shaunharrisonFR**
Add in readmes with installation instructions

Issue: https://github.com/SecureApiGateway/SecureApiGateway/issues/939

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,104 @@
-# secure-api-gateway-releases
-This repo holds the versioned helm and kustomize objects to be able to deploy the third party sources and secure-api-gateway
+# Secure API Gateway - Releases
 
+## Prerequisites
```
Comment (BohoCode) [inline README.md:3]: Could do with an Overview section here. It should answer the following questions. 

Why does this repository exist?
What was your motivation?
Why did you build this project?
What problem does it solve?

File: README.md [hunk 2]
```diff
@@ -1,3 +1,104 @@
...
+It is assumed that, either a deployments repo has been created containing the specific deployment overrides has been created, or if a repository are not being used they are at hand (within the instructions below it is assumed that a repo is being used)
+
+It is assumed that all infrastructure has been created, including deploying the Cloud Instance, and that the following prerequisits have been deployed to the cluster
+- [Nginx](https://kubernetes.github.io/ingress-nginx/deploy/)
+- [MongoDB](https://github.com/SecureApiGateway/secure-api-gateway-releases/blob/master/third-party/Chart.yaml)
+- [Secrets](https://github.com/SecureApiGateway/secure-api-gateway-releases-helpers/tree/master/external-secrets-gsm) 
+  NOTE: GSM has been used but is not manditory, other secrets managers can be used.
+
+## Deployment Guide [Internal]
+
```
Comment (BohoCode) [inline README.md:33]: This should be documented in the internal wiki so as not to confuse 'customers' of the project. Really, you're documenting how to use the Action, so by all means put a README in github workflows.

File: README.md [hunk 3]
```diff
@@ -1,3 +1,104 @@
...
+Once the workflow has completed, a new tag and release will have been created within secure-api-gateway-release repo, and new helm packages will have been pushed to the atrifactory.
+
+| Key | Description | Example |
+|-----|-------------|---------|
+| `VERSION` | Version of the charts to publish, must be unique and not already be in use | 1.0.0 |
+
+## Installation Guide
+
+### Assumptions
+At this point all infrastructure has been installed, a cloud instance has been created and configured, secrets have been created and deployed to kubernetes cluster and the softwares have been built with docker images deployed to a container registry.
```
Comment (BohoCode) [inline README.md:53]: Links to the overview of what kind of infrastructure might be needed to run the project?

File: secure-api-gateway-helpers/readme.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
+# Secure API Gateway - Releases - Secure API Gateway Helpers
+
+## Prerequisites
```
Comment (BohoCode) [inline secure-api-gateway-helpers/readme.md:3]: Overview section again please!

**Review by BohoCode — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (shaunharrisonFR) [reply README.md:22]: Added in generic introduction & repo specific overview

File: secure-api-gateway-helpers/readme.md (hunk 4 above)
Comment (shaunharrisonFR) [reply secure-api-gateway-helpers/readme.md:3]: Will do this one once the 'Main' Readme has been okay'd

File: README.md (hunk 2 above)
Comment (shaunharrisonFR) [reply README.md]: Done! Moved to Readme in workflows

File: README.md (hunk 3 above)
Comment (shaunharrisonFR) [reply README.md]: Linked to placeholder wiki page

File: README.md [hunk 5]
```diff
@@ -1,3 +1,110 @@
...
+NOTE: For Secrets, Google Secrets Manager has been used but is not manditory, other secrets managers can be used.
+
+## Installation Guide
+
+### Assumptions
+At this point all [infrastructure](https://github.com/SecureApiGateway/SecureApiGateway/wiki/Recommended-Infrastructure) has been installed, a cloud instance has been created and configured, secrets have been created and deployed to kubernetes cluster and the softwares have been built with docker images deployed to a container registry. 
+
+These steps can be performed manually or via a pipeline (within the instructions below it is assumed that these are being performed manually)
+
+It is assumed that the structure of the deployment repo looks like the following
```
Comment (BohoCode) [inline README.md:59]: If I were a customer reading this I would be asking "What deployment repo"?

We probably need to write an overview document in the wiki for how to approach deploying all of this, then we can fill in details in the READMEs and also refer to the overview document from the wikis?

**Review by BohoCode — CHANGES_REQUESTED** [review]

File: README.md [hunk 6]
```diff
@@ -1,3 +1,110 @@
...
+
+## Repo Overview
+
+The purpose of this repository is to provide customers with installable versions of the secure-api-gateway.
+Customers will be able to take the releases of this repo for version 1,2,N and deploy to a kubernetes infrastructure.
+The secure-api-gateway involves a number of components which are installed via [helm](https://helm.sh/) and [kustomize](https://kustomize.io/) and overlays which will all be documented here.
+
+The issue that this repository is trying to tackle is that the components are all in individual repositories, within the repositories are the infrastructure objects files. Customers previously would have to deploy from the individual repositories, ensuring that the values.yaml files are modified for their deployment. This would be difficult if mutliple versions of the secure-api-gateway are being managed as the values.yaml file would need modifying each time. 
+
+With this repository we are providing a set of versions of the secure api gateway, meaning that a customer can deploy numerous verisons of V1.0.0, V1.0.1, V2.0.0 ect. A benefit of this is that the custoemer no longer needs to figure out which components are compatible with each other.
```
Comment (BohoCode) [inline README.md:15]: sp custoemer

**Review by BohoCode — CHANGES_REQUESTED** [review]

**Review by BohoCode — APPROVED** [review]

**Pull request merged**